client/client.go

```diff
@@ -402,7 +406,49 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	return client.doSyncWithOpts(method, path, query, headers, body, v, nil)
 }
 
+func (client *Client) checkMaintenanceJSON() error {
+	f, err := os.Open(dirs.SnapdMaintenanceFile)
+	// just continue if we can't read the maintenance file and
+	// it does exist (i.e. perm error)
+	if err == nil {
+		// we have a maintenance file, try to read it
+		maintenance := &Error{}
+
+		if err := json.NewDecoder(f).Decode(&maintenance); err != nil {
+			// TODO: should we just let the request go through even if the
+			// maintenance.json is unintelligible?
+			return err
+		}
+
+		if maintenance != nil {
+			switch maintenance.Kind {
+			case ErrorKindDaemonRestart:
+				client.maintenance = maintenance
+			case ErrorKindSystemRestart:
+				client.maintenance = maintenance
+			}
+			// don't set maintenance for other kinds, as we don't know what it
+			// is yet
+
+			// this also means an empty json object in maintenance.json doesn't get
+			// treated as a real maintenance downtime for example
```

Comment:
I don't mind this much but I think it would be simpler to say that an empty maintenance file is equivalent to no maintenance at all. Then we can only deserialize non-empty files.

Reply:
that seems fine with me

Reply:
sure sounds good